src/common/ipc.ts

```diff
@@ -2,94 +2,133 @@
 // https://www.electronjs.org/docs/api/ipc-main
 // https://www.electronjs.org/docs/api/ipc-renderer
 
-import { ipcMain, ipcRenderer, WebContents, webContents } from "electron"
+import { ipcMain, ipcRenderer, IpcRendererEvent, WebContents, webContents } from "electron"
 import logger from "../main/logger";
+import { getRandId } from "./utils";
 
 export type IpcChannel = string;
 
-export interface IpcHandleOpts {
-  timeout?: number;
+export enum IpcMode {
+  SYNC = "sync",
+  ASYNC = "async",
 }
 
-export interface IpcMessageHandler<T extends any[] = any> {
-  (...args: T): any;
+export interface IpcChannelRequest<A extends any[] = any> {
+  msgId: string;
+  args: A;
 }
 
-export interface IpcMessageOpts<A extends any[] = any> {
+export interface IpcChannelResponse<T extends any[] = any, E = any> {
+  msgId: string;
+  data?: T;
+  error?: E;
+}
+
+export interface IpcChannelInit {
```

Comment:
This doesn't seem to be about the initial channel, wouldn't something like `IpcChannelOpts` be more fitting?

Reply:
yep, sounds good